src/Mod/Draft/draftviewproviders/view_dimension.py

```diff
@@ -572,9 +572,18 @@ def updateData(self, obj, prop):
         # Special representation if we use 'Building US' scheme
         u_params = App.ParamGet("User parameter:BaseApp/Preferences/Units")
         if u_params.GetInt("UserSchema", 0) == 5:
-            s = App.Units.Quantity(length, App.Units.Length).UserString
-            self.string = s.replace("' ", "'- ")  # feet
-            self.string = s.replace("+", " ")
+            self.string = App.Units.Quantity(length, App.Units.Length).UserString
+            if self.string.count('"') > 1:
+                # multiple inch tokens
+                self.string = self.string.replace('"',"",self.string.count('"')-1)
+            d_params = App.ParamGet("User parameter:BaseApp/Preferences/Mod/Draft")
+            sep = d_params.GetString("FeetSeparator","")
+            if sep:
+                # use a custom separator
+                self.string = self.string.replace("' ", "'" + sep + " ")
```

Comment:
It may be undesirable to have a space after the user specified FeetSeparator?

Reply:
Hm you're right... probably better to either use " " as default, or let the user compose their own, spaces included.

Reply:
One check seems to fail. You may need to rebase to trigger new checks?